include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,401 @@
+            : current_(std::move(i.current_)) {}
+
+    private:
+        template <typename... Iterators>
+        constexpr iterator(const Iterators&... iterators)
+            : current_(iterators...) {}
+    public:
+
+        constexpr return_tuple_type operator*() const {
+            auto __tr = [](auto&&... __args) -> decltype(auto)
+                { return return_tuple_type(std::forward<decltype(__args)>(__args)...);};
+            return apply_to_tuple(__tr, [](auto it) -> decltype(auto) { return *it;}, current_);
+        }
+
+        constexpr return_tuple_type operator[]( difference_type n ) const
+            requires all_random_access<Const, Views...>
+        {
+            return *(*this + n);
+        }        
+
+        constexpr iterator& operator++() {
+            zip_view::apply_to_tuple([](auto& it) { return ++it; }, current_);
+            return *this;
+        }
+
+        constexpr void operator++(int) {
+            ++*this;
+        }
+
+        constexpr iterator operator++(int) requires all_forward<Const, Views...> {
+            auto tmp = *this;
+            ++*this;
+            return tmp;
+        }
+
+        constexpr iterator& operator--() requires all_bidirectional<Const, Views...> {
+            zip_view::apply_to_tuple([](auto& it) { return --it; }, current_);
+            return *this;
+        }
+
+        constexpr iterator operator--(int) requires all_bidirectional<Const, Views...> {
+            auto tmp = *this;
+            --*this;
+            return tmp;
+        }
+
+        constexpr iterator& operator+=(difference_type n)
+            requires all_random_access<Const, Views...>
+        {
+            zip_view::apply_to_tuple([n](auto& it) { return it += n; }, current_);
+            return *this;
+        }
+
+        constexpr iterator& operator-=(difference_type n)
+            requires all_random_access<Const, Views...>
+        {
+            zip_view::apply_to_tuple([n](auto& it) { return it -= n; }, current_);
+            return *this;
+        }
+
+        friend constexpr bool operator==(const iterator& x, const iterator& y)
+            requires ( std::equality_comparable<std::conditional_t<!Const, std::ranges::iterator_t<Views>,
+                                                                           std::ranges::iterator_t<const Views>>> && ... )
+        {
+            if constexpr (all_bidirectional<Const, Views...>) {
+                return x.current_ == y.current_;
+            } else {
+                return x.compare_equal(y, std::make_index_sequence<sizeof...(Views)>());
+            }
+        }
+
+        friend constexpr auto operator<=>(const iterator& x, const iterator& y)
+            requires all_random_access<Const, Views...>
+        {
+            if (x.current_ < y.current_)
+                return -1;
+            else if (x.current_ == y.current_)
+                return 0;
+            return 1; //x.current > y.current_
```

Comment:
When adding comparison between internal tuples, we looked at adding spaceship and it was deemed not worth the effort at the time.  We could use this as motivation for adding it, but it would require a bit of work.  We would need to trigger it off of `__cpp_lib_three_way_comparison >= 201907L`, and in that case replace the existing comparison operators with the spaceship, to time it properly with std::tuple <=> operator

https://github.com/oneapi-src/oneDPL/pull/1472#issuecomment-2032405450

Reply:
The thought was that it wasn't worth the maintenance because we would need to implement both to support our full support matrix, and it would be only helpful for those explicitly calling `<=>` on the internal tuple (which is what we might want to do here as I understand it).